---
title: "GhanaCovid-19SequencingSummary"
author: "WACCBIP"
date: "6/3/2021"
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
---

```{r setup, include=FALSE}
pacman::p_load(scran, patchwork, viridis, ggforce, gghalves, ggridges, scDblFinder, SingleR, intrinsicDimension, cluster, data.table, ggnetwork, ggplot2, tibble, tidyverse, dplyr,Seurat,glmGamPoi, devtools,reticulate,stringr,magrittr,ComplexHeatmap,circlize,viridis,svglite,liger,grr, Matrix.utils, comprehenr, PNWColors,  scater, pheatmap, ExperimentHub, AnnotationHub, SeuratData, SeuratDisk, ggsignif, GGally, RColorBrewer, MAST, slingshot, tradeSeq, clusterExperiment, ggpubr, clustree,phateR,mgcv, ComplexHeatmap, SingleCellExperiment, phateR, limma)

pacman::p_load(plyr, ggplot2, gplots, grid, spatstat, raster, sp, dplyr, klaR, ggfortify, stringr, cluster, Rtsne, readr, RColorBrewer, Hmisc, mice, tidyr, purrr, VIM, magrittr, corrplot, caret, gridExtra, ape, tidytree, pheatmap, RColorBrewer, stats, vegan, FactoMineR, factoextra, outliers, ggpubr, table1,ggvis, ggsignif, qdap, wordcloud2, arsenal, knitr)

knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(janitor)
library("readxl")
library(cowplot)
library(ggthemes)
library(ggrepel)
library(ggalluvial)
library("lubridate")
library(ggtree)
library(ape)
library(treeio)
library(scales) # to access breaks/formatting functions
library(ggtreeExtra)
library(ggstar)
library(ggnewscale)
library(pier)
library(table1)


custom_colors <- list()
colors_dutch <- c(
  '#2484c1','#86f71a','#cc9fb1','#D980FA','#F79F1F',
  '#EE5A24','#009432','#833471','#1B1464','#0652DD',
  '#006266','#EA2027','#1B1464','#5758BB','#6F1E51'
)
colors_spanish <- c(
  '#40407a','#706fd3','#f7f1e3','#34ace0','#33d9b2',
  '#2c2c54','#474787','#aaa69d','#227093','#218c74',
  '#ff5252','#ff793f','#d1ccc0','#ffb142','#ffda79',
  '#b33939','#cd6133','#84817a','#cc8e35','#ccae62'
)
custom_colors$discrete <- c(colors_dutch, colors_spanish)
custom_colors$cell_cycle <- setNames(
  c('#45aaf2', '#f1c40f', '#e74c3c', '#7f8c8d'),
  c('G1',      'S',       'G2M',     '-')
)

##Formats
niceFormat <- function(number) {
  formatC(number, format = 'f', big.mark = ',', digits = 0)
}


theme_USGS_box <- function(base_family = "", ...){
  theme_bw(base_family = base_family, ...) +
    theme(
      panel.grid = element_blank(),
      plot.title = element_text(size = 8, hjust = 0.5),
      axis.ticks.length = unit(-0.05, "in"),
      axis.text.y = element_text(margin=unit(c(0.3,0.3,0.3,0.3), "cm")),
      axis.text.x = element_text(margin=unit(c(0.3,0.3,0.3,0.3), "cm")),
      axis.ticks.x = element_blank(),
      #aspect.ratio = 1,
      legend.background = element_rect(color = "black", fill = "white"),
      text = element_text(size = 12, colour = "black"),
      plot.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = 'pt')
    )}
```
Sequencing Report for isolates collected in Ghana
Summary Tables
```{r, echo=FALSE, warning=FALSE, comment=TRUE}

# # ##Read in the lineage file
# pangoNextErrInser <- read.csv("/Users/collinsmisita/manuscript/009_merged_lineage_reports/PangoNext_all_runs.csv")
# #metadata
# Metadata2 <- read.csv("/Users/collinsmisita/OneDrive - University of Ghana/007_Waccbip_Manuscripts/Covid-2021_manuscript/NatureComm/ENA_DATA_Submission/RawData_SARS-CoV-2-Manuscript-Ghana_prep3.csv")
#  table <- table(Metadata2$Curated.month.year)
# #Merge them
# MetaPangoNext <-merge(Metadata2, pangoNextErrInser, by.x ="Run_barcode_ID", by.y = "UniqueID", all.x = TRUE, all.y= FALSE)
# 
# #Read in the files
# genbank <- read.csv("/Users/collinsmisita/manuscript/013_Genbank_official_Report/Genbank_submission_Report.csv", header = T, na.strings = T) ; glimpse(genbank)
# gisaid <- read.csv("/Users/collinsmisita/manuscript/011_gisaid_official_reports/gisaid_hcov-19_2021_12_07_14.csv", header = T, na.strings = T) ; glimpse(gisaid)
# 
# #Merge the datasets
# genbaGisaid <- merge(gisaid,genbank, by.x ="Virus_name", by.y = "Sequence.ID", all.x = TRUE, all.y= TRUE); glimpse(genbaGisaid)
# 
# MetaPangoNext <- merge(MetaPangoNext,genbaGisaid, by.x ="Virus.name", by.y = "Virus_name", all.x = TRUE)
# 
# # ##View all available data variables
# # 
# MetaPangoNext <- MetaPangoNext %>% mutate(agegroup = case_when(
#   age >= NA ~ 'NA', 
#   age >= 91  & age <= 100 ~ '91-100 Years',
#   age >= 81  & age <= 90 ~ ' 81-90 Years',age >= 71  & age <= 80 ~ ' 71-80 Years',
#   age >= 61  & age <= 70 ~ ' 61-70 Years',age >= 51  & age <= 61 ~ ' 51-60 Years',
#   age >= 41  & age <= 50 ~ ' 41-50 Years',age >= 31  & age <= 40 ~ ' 31-40 Years',
#   age >= 21  & age <= 30 ~ ' 21-30 Years',age >= 11  & age <= 20 ~ ' 11-20 Years',
#   age >= 0  & age <= 10 ~ '  0-10 Years')) # end function
# table <- table(MetaPangoNext$agegroup)
# table %>% knitr::kable(caption = "Categorizing Age into groups (n)")
# # 
# # 
# MetaPangoNext$Variant <- gsub("B.1.1.7","B.1.1.7 (Alpha)",MetaPangoNext$lineage, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("B.1.525","B.1.525 (Eta)",MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("B.1.351","B.1.351 (Beta)",MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("B.1.617.1","B.1.617.1 (Kappa)",MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("B.1.617.2","B.1.617.2 (Delta)",MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("B.1.526","B.1.526 (Iota)",MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)
# MetaPangoNext$Variant <- gsub("AY.*","AY.* (Delta Plus)",MetaPangoNext$Variant, perl = TRUE); table(MetaPangoNext$Variant)


# MetaPangoNext$VariantType <- gsub("B.1.1.7 (Alpha)","VOC",MetaPangoNext$Variant, fixed = TRUE)
# MetaPangoNext$VariantType <- gsub("B.1.525 (Eta)","VOC", MetaPangoNext$VariantType, fixed = TRUE)
# MetaPangoNext$VariantType <- gsub("B.1.617.2 (Delta)","VOC",MetaPangoNext$VariantType, fixed = TRUE)
# MetaPangoNext$VariantType <- gsub("B.1.1.318","VOC",MetaPangoNext$VariantType, fixed = TRUE)
# MetaPangoNext$VariantType <- gsub("B.1.1","VOC",MetaPangoNext$VariantType, fixed = TRUE) 
# 
# MetaPangoNext <- MetaPangoNext %>% mutate(VariantType = case_when(VariantType != "VOC" ~ "Non VOC",TRUE ~ as.character(VariantType)))

#Write to csv

MetaPangoNext <- read.csv("/Users/collinsmisita/OneDrive - University of Ghana/007_Waccbip_Manuscripts/Covid-2021_manuscript/NatureComm/ENA_DATA_Submission/RawData_14_Jan_2021_EditedCT.csv", header = T)

MetaPangoNext <- MetaPangoNext %>% mutate(agegroup = case_when(
  age >= NA ~ 'NA',
  age >= 61  & age <= 99 ~ ' 61-100 Years',
  age >= 41  & age <= 60 ~ ' 41-60 Years',
  age >= 21  & age <= 40 ~ ' 21-40 Years',
  age >= 0  & age <= 20 ~ '  0-20 Years')) # end function
table <- table(MetaPangoNext$agegroup)

MetaPangoNext <- MetaPangoNext %>% mutate(agegroup2 = case_when(
  age >= NA ~ 'NA',
  age >= 50  & age <= 99 ~ ' 51+ Years',
  age >= 25  & age <= 50 ~ ' 25-50 Years',
  age >= 0  & age <= 25 ~ '  <25 Years')) # end function
table <- table(MetaPangoNext$agegroup)

table(MetaPangoNext$VariantType)

MetaPangoNext$Curated.month.year <- factor(
  MetaPangoNext$Curated.month.year,levels = c( "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021", "Unknown"))

MetaPangoNext$Variant <- gsub("AY.* (Delta Plus)", "B.1.617.2 (Delta)", MetaPangoNext$Variant, fixed = TRUE); table(MetaPangoNext$Variant)

MetaPangoNext_QC <- MetaPangoNext
MetaPangoNext_traveller <- MetaPangoNext_QC %>% filter(Travel_info == "Traveler" | Travel_info == "UHAS-Traveler")
MetaPangoNext_Local_T <- MetaPangoNext_QC
MetaPangoNext_Local <- MetaPangoNext_QC %>% filter(Travel_info == "Local" | Travel_info == "UHAS") 
MetaPangoNext_2021 <- MetaPangoNext_Local %>% filter(Curated.Year.b == "2021")
MetaPangoNext_2020 <- MetaPangoNext_Local %>% filter(Curated.Year.b == "2020")
MetaPangoNext_deltaplus <- MetaPangoNext_Local %>% filter(Variant == "AY.* (Delta Plus)")
#QC Report

write.csv(MetaPangoNext, "/Users/collinsmisita/OneDrive - University of Ghana/007_Waccbip_Manuscripts/Covid-2021_manuscript/NatureComm/ENA_DATA_Submission/RawData_14_Jan_2021.csv")

MetaPangoNext <- MetaPangoNext %>% mutate(
  waccbipQc = case_when(totalMissing >= 15000 ~ 'Discard',
                        totalMissing >= 4000  & totalMissing <= 14999 ~ ' RetainForVariantsCount',
                        totalMissing >=  500 & totalMissing <= 3999 ~ ' AcceptableForSubmission',
                        totalMissing >= 0  & totalMissing <= 499 ~ '  Good')) # end function
table <- table(MetaPangoNext$waccbipQc)
table %>% knitr::kable(caption = "Genome Length") 
median(MetaPangoNext$totalMissing)

 table <- table(MetaPangoNext_Local$Curated.Region)
 table %>% knitr::kable(caption = "No. of Genomes by sampling Regions")
#
 
 table <- table(MetaPangoNext$Curated.Year.b)
 table %>% knitr::kable(caption = "No. of Genomes by sampling month and year")
 
 table <- table(MetaPangoNext_Local$Curated.month.year)
 table %>% knitr::kable(caption = "No. of Genomes by sampling month and year")
 
```

2) Overall Summary for all the Ghana Genomes (Both Locals and Travelers)

```{r, echo=FALSE, warning=FALSE}
plot_counts <- MetaPangoNext_Local_T %>%  dplyr::group_by(Variant, Travel_info) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Travel_info, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


#Convert all Variants below 4 to other Variants
#Subset data for various plots
plot_counts_voc <- plot_counts %>% select(c("Variant", "Local", "Traveler", "Total")) %>% filter(Variant %in% c("B.1.1.7 (Alpha)", "B.1.351 (Beta)", "B.1.617.2 (Delta)","B.1.525 (Eta)","B.1.617.1 (Kappa)","B.1.526 (Iota)"))
plot_counts_others <- plot_counts %>% select(c("Variant", "Local", "Traveler", "Total")) %>% filter(Total <= 6)
plot_counts <- plot_counts %>% select(c("Variant", "Local", "Traveler", "Total")) %>% mutate(Variant = case_when(Total <= 6 ~ "Other Variants",TRUE ~ as.character(Variant)))

#Summarize the Variants
plot_counts1 <- plot_counts %>% filter(Variant != "Other Variants")
plot_counts2 <- plot_counts %>% filter(Variant == "Other Variants")
plot_counts3 <- plot_counts2 %>% group_by(Variant) %>% summarise(across(Local:Total, sum))
plot_counts4 <- dplyr::bind_rows(plot_counts1,plot_counts3)
plot_counts5 <- plot_counts4 %>% summarise(across(Local:Total, sum)); rownames(plot_counts5)[rownames(plot_counts5) == "1"] <- "Grand Total"
plot_counts6 <- column_to_rownames(plot_counts4, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts6,plot_counts5)
plot_counts <- rownames_to_column(plot_counts7, "Variant")
plot_counts_Final <- plot_counts 
plot_counts_Final %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Plot pie chart for all samples
plot_counts <- plot_counts4
plot_counts$color <- NA
plot_counts$value <- plot_counts$Total
plot_counts$label <- plot_counts$Variant
plot_counts <- plot_counts %>%select(-c('Total', "Local", "Variant")) 
advanced.pie <- plot_counts %>% pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Lineages', font='Impact', location='pie-center') %>%
  pie.subtitle(text='Overall (n=1123)') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>%
  pie.tooltips() %>%  pie.labels(text = element_text(size = 18))
advanced.pie

#Plot Table for the others
plot_countsX <- plot_counts_others %>% summarise(across(Local:Total, sum))
rownames(plot_countsX)[rownames(plot_countsX) == "1"] <- "Grand Total" 
plot_countsX2 <- rownames_to_column(plot_countsX, "Variant")
plot_countsX3 <- dplyr::bind_rows(plot_counts_others,plot_countsX2)
plot_countsX3 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


#Plot Table for VOCs
plot_countsZ <- plot_counts_voc %>% summarise(across(Local:Total, sum))
rownames(plot_countsZ)[rownames(plot_countsZ) == "1"] <- "Grand Total" 
plot_countsZ2 <- rownames_to_column(plot_countsZ, "Variant")
plot_countsZ3 <- dplyr::bind_rows(plot_counts_voc,plot_countsZ2)
plot_countsZ3 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#plot pie chart
plot_counts <- plot_counts_voc
plot_counts$color <- NA
plot_counts$value <- plot_counts$Total
plot_counts$label <- plot_counts$Variant
plot_counts <- plot_counts %>%select(-c('Total', "Local", "Variant")) 
advanced.pie <- plot_counts %>% pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Variants', font='Impact', location='pie-center') %>%
  pie.subtitle(text='VOC - Overall') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>% pie.tooltips()
advanced.pie

```


2) Travelers

```{r, echo=FALSE, warning=FALSE}

#plot Table for local samples only
plot_counts <- plot_counts_Final %>% select(-c("Local", "Total"))
plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Plot pie chart for Traveler samples
plot_counts <- plot_counts4
plot_counts$color <- NA
plot_counts$value <- plot_counts$Traveler
plot_counts$label <- plot_counts$Variant
advanced.pie <- plot_counts %>% pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Lineages', font='Impact', location='pie-center') %>%
  pie.subtitle(text='Travelers (n=106)') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>%
  pie.tooltips() %>%  pie.labels(text = element_text(size = 18))
advanced.pie

#Plot Table for the others
plot_counts_Traveler <- plot_countsX3 %>% select(-c("Local", "Total")) %>% filter(Traveler != 0)
#plot_counts_Traveler %>%  knitr::kable(caption = "Other Variants (Overall) detected in Ghana")

##Plot VOC for traveller
plot_counts <- plot_counts_voc
plot_counts$color <- NA
plot_counts$value <- plot_counts$Traveler
plot_counts$label <- plot_counts$Variant
advanced.pie <- plot_counts %>% pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Lineages', font='Impact', location='pie-center') %>%
  pie.subtitle(text='VOC - Traveler Samples') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>% pie.tooltips()
advanced.pie
```

2) Locals

```{r, echo=FALSE, warning=FALSE}
#plot Table for local samples only
plot_counts <- plot_counts_Final %>% select(-c("Traveler", "Total"))
plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Plot pie chart for Traveler samples
plot_counts <- plot_counts4
plot_counts$color <- NA
plot_counts$value <- plot_counts$Local
plot_counts$label <- plot_counts$Variant

advanced.pie <- plot_counts %>%
  pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Lineages', font='Impact', location='pie-center') %>%
  pie.subtitle(text='Local Samples') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>%
  pie.tooltips() %>% 
  pie.labels(text = element_text(size = 18))
advanced.pie

#Plot Table for the others
plot_counts_local <- plot_countsX3 %>% select(-c("Traveler", "Total")) %>% filter(Local != 0)
#plot_counts_local %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

##Plot VOC for traveller
plot_counts <- plot_counts_voc
plot_counts$color <- NA
plot_counts$value <- plot_counts$Local
plot_counts$label <- plot_counts$Variant
advanced.pie <- plot_counts %>%
  pier() %>%
  pie.size(inner=70, outer=100, width = 600, height = 450) %>%
  pie.header(text='Pango Lineages', font='Impact', location='pie-center') %>%
  pie.subtitle(text='VOC - Local Samples') %>%
  pie.footer(text='Dynamic', location = 'bottom-left') %>%
  pie.tooltips()
advanced.pie

```

5) Variant Summary across Month and Year for Local Genomes Only

```{r, echo=FALSE, warning=FALSE, comment=TRUE}
###Month Clade Comparison swapped

plot_counts <- MetaPangoNext_Local %>%  dplyr::group_by(Variant, Curated.month.year) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Curated.month.year, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")
#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 7 ~ "Other Variants", TRUE ~ as.character(Variant)))


#Plot the summarized table
plot_counts1 <- plot_counts %>% filter(Variant != "Other Variants")
plot_counts2 <- plot_counts %>% filter(Variant == "Other Variants")
plot_counts3 <- plot_counts2 %>% group_by(Variant) %>% summarise(across(Total:`Aug, 2021`, sum))
plot_counts4 <- dplyr::bind_rows(plot_counts1,plot_counts3)
plot_counts5 <- plot_counts4 %>% summarise(across(Total:`Aug, 2021`, sum)); rownames(plot_counts5)[rownames(plot_counts5) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts5, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts4,plot_counts6) %>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021", "Total")# "Jul, 2021",
plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


plot_counts7 <- plot_counts4 %>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021")# "Jul, 2021",
row.names(plot_counts7) <- plot_counts7$Variant; #plot_counts7$Variant <- NULL;
plot_counts8 <- plot_counts7 %>% select(-c("Variant"))
plot_counts8 <- as.matrix(plot_counts8)
data3d <- prop.table(plot_counts8, 2)
data3d <- as.data.frame(data3d)

#Generate plots
p1 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  #scale_fill_manual(name = 'Variant(Local)', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
    scale_fill_manual(name = 'Variant', values = c("B.1.617.2 (Delta)" = "#2484c1", "B.1.1.7 (Alpha)" = "chartreuse3",
                                                  "B.1.1.318" = "plum4","B.1.1" = "salmon4","B.1.525 (Eta)" = "red4",
                                                  "B.1.1.359" = "yellow3", "B.1"= "darkorange","B.1.623"="sienna3", 
                                                  "A" = "lightgoldenrod4","B.1.1.485" = "seagreen1",
                                                  "B.1.351 (Beta)" = "blue", "A.11"= "darkolivegreen4",
                                                  "Other Variants" = "darkgreen")) +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() +
  #ggtitle("B) Dynamics of gisaid Variants from 2020 to 2021-Locals") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt')); p1
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Curated.Month.Year_Variant_all.svg', p1, width = 6, height = 4)


plot_counts <- MetaPangoNext_traveller %>%  dplyr::group_by(Variant, Curated.month.year) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Curated.month.year, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
#plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 6 ~ "Other Variants",TRUE ~ as.character(Variant)))

#Plot summarized Table
#Plot the summarized table
#Plot the summarized table
plot_counts1 <- plot_counts %>% filter(Variant != "Other Variants")
plot_counts2 <- plot_counts %>% filter(Variant == "Other Variants")
plot_counts3 <- plot_counts2 %>% group_by(Variant) %>% summarise(across(Total:`Mar, 2021`, sum))
plot_counts4 <- dplyr::bind_rows(plot_counts1,plot_counts3)
plot_counts5 <- plot_counts4 %>% summarise(across(Total:`Mar, 2021`, sum)); rownames(plot_counts5)[rownames(plot_counts5) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts5, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts4,plot_counts6) %>% select("Variant",  "Jan, 2021", "Mar, 2021", "Total")
#plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Generate plots
p2 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  scale_fill_manual(name = 'Variant (Travelers)', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() +
  ggtitle("B) Dynamics of gisaid Variants from 2020 to 2021 - Travellers") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt')); #p2
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Curated.Month.Year_Variant_Traveler.svg', p2 , width = 3, height = 4)


##Table for country of Origin
plot_counts <- MetaPangoNext_traveller %>%  dplyr::group_by(Variant, Curated.month.year, Travel_History) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Curated.month.year, count, fill = 0)# %>% dplyr::ungroup() %>%  dplyr::select(c("Variant", dplyr::everything())) #%>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Travel History")

##Table for country of Origin
plot_counts <- MetaPangoNext_traveller %>%  dplyr::group_by(Curated.month.year,  Travel_History) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Travel_History, count, fill = 0) %>% dplyr::ungroup() %>%  dplyr::select(c("Curated.month.year", dplyr::everything())) #%>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Travel History")

plot_counts <- MetaPangoNext_traveller %>%  dplyr::group_by(Variant, Curated.month.year) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Curated.month.year, count, fill = 0) %>% dplyr::ungroup() %>%  dplyr::select(c("Variant", dplyr::everything())) #%>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Travel History")


```


6) Variant Summary by Sampling Region in Ghana

```{r, echo=FALSE,warning=FALSE}

###Location & Variant Comparison
plot_counts <- MetaPangoNext_2021 %>%  dplyr::group_by(Variant, Curated.Region) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Curated.Region, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
#plot_counts %>% knitr::kable(caption = "Gisaid Variants by Regions - Local samples (2021)")

plot_counts3 <- plot_counts %>% summarise(across(Total:`Western`, sum))
rownames(plot_counts3)[rownames(plot_counts3) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts3, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts,plot_counts6) %>% select("Variant", "Ashanti", "Bono East", "Central", "Eastern", "Greater Accra", "Northern", "Upper East", "Volta", "Western", "Total")
plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 6 ~ "Other Variants",TRUE ~ as.character(Variant)))

#Generate plot
p1 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  #scale_fill_manual(name = 'Variant', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
    scale_fill_manual(name = 'Variant', values = c("B.1.617.2 (Delta)" = "#2484c1", "B.1.1.7 (Alpha)" = "chartreuse3",
                                                  "B.1.1.318" = "plum4","B.1.1" = "salmon4","B.1.525 (Eta)" = "red4",
                                                  "B.1.1.359" = "yellow3", "B.1"= "darkorange","B.1.623"="sienna3", 
                                                  "A" = "lightgoldenrod4","B.1.1.485" = "seagreen1",
                                                  "B.1.351 (Beta)" = "blue", "A.11"= "darkolivegreen4",
                                                  "Other Variants" = "darkgreen")) +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() +
  ggtitle("Variants by regions (2021)") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), 
        plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt'))
p1

ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Curated.Region_Variant_2021.svg', 
       p1, width = 5, height = 5)

###Location & Variant Comparison
plot_counts <- MetaPangoNext_2020 %>%  dplyr::group_by(Variant, Curated.Region) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Curated.Region, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))

plot_counts3 <- plot_counts %>% summarise(across(Total:`Western`, sum))
rownames(plot_counts3)[rownames(plot_counts3) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts3, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts,plot_counts6) %>% select("Variant", "Ashanti", "Central", "Eastern", "Greater Accra", "Volta", "Western", "Total")
plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")

#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 4 ~ "Other Variants",TRUE ~ as.character(Variant)))

#Generate plot
p2 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  #scale_fill_manual(name = 'Variant', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
    scale_fill_manual(name = 'Variant', values = c("B.1.617.2 (Delta)" = "#2484c1", "B.1.1.7 (Alpha)" = "chartreuse3",
                                                  "B.1.1.318" = "plum4","B.1.1" = "salmon4","B.1.525 (Eta)" = "red4",
                                                  "B.1.1.359" = "yellow3", "B.1"= "darkorange","B.1.623"="sienna3", 
                                                  "A" = "lightgoldenrod4","B.1.1.485" = "seagreen1",
                                                  "B.1.351 (Beta)" = "blue", "A.11"= "darkolivegreen4",
                                                  "Other Variants" = "darkgreen")) +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() +
  ggtitle("Variants by regions (2020)") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt'))
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Curated.Region_Variant_2020.svg', 
       p2 , width = 5, height = 5)
p2


#######################################################################################################################
```

Age Groups
```{r}
###Month Clade Comparison swapped

plot_counts <- MetaPangoNext_Local %>%  dplyr::group_by(Variant, agegroup) %>% dplyr::summarize(count = n()) %>% tidyr::spread(agegroup, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
plot_counts %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")
#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 7 ~ "Other Variants", TRUE ~ as.character(Variant)))


#Plot the summarized table
plot_counts1 <- plot_counts %>% filter(Variant != "Other Variants")
plot_counts2 <- plot_counts %>% filter(Variant == "Other Variants")
plot_counts3 <- plot_counts2 %>% group_by(Variant) %>% summarise(across(Total:`<NA>`, sum))
plot_counts4 <- dplyr::bind_rows(plot_counts1,plot_counts3)
plot_counts5 <- plot_counts4 %>% summarise(across(Total:`<NA>`, sum)); rownames(plot_counts5)[rownames(plot_counts5) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts5, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts4,plot_counts6) #%>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021", "Total")# "Jul, 2021",
plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


plot_counts7 <- plot_counts4 #%>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021")# "Jul, 2021",
row.names(plot_counts7) <- plot_counts7$Variant; #plot_counts7$Variant <- NULL;
plot_counts8 <- plot_counts7 %>% select(-c("Variant"))
plot_counts8 <- as.matrix(plot_counts8)
data3d <- prop.table(plot_counts8, 2)
data3d <- as.data.frame(data3d)

#Generate plots
plot_counts <- plot_counts %>% rename("Unknown" = `<NA>`)

p1 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  #scale_fill_manual(name = 'Variant(Local)', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
    scale_fill_manual(name = 'Variant', values = c("B.1.617.2 (Delta)" = "#2484c1", "B.1.1.7 (Alpha)" = "chartreuse3",
                                                  "B.1.1.318" = "plum4","B.1.1" = "salmon4","B.1.525 (Eta)" = "red4",
                                                  "B.1.1.359" = "yellow3", "B.1"= "darkorange","B.1.623"="sienna3", 
                                                  "A" = "lightgoldenrod4","B.1.1.485" = "seagreen1",
                                                  "B.1.351 (Beta)" = "blue", "A.11"= "darkolivegreen4",
                                                  "Other Variants" = "darkgreen")) +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() +
  #ggtitle("B) Dynamics of gisaid Variants from 2020 to 2021-Locals") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt')); p1
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Variant_distribution across Age groups.svg', p1, width = 6, height = 4)
```

Gender

```{r}
###Month Clade Comparison swapped
plot_counts <- MetaPangoNext_Local %>%  dplyr::group_by(Variant, Gender) %>% dplyr::summarize(count = n()) %>% tidyr::spread(Gender, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("Variant", "Total", dplyr::everything())) %>% arrange(desc(Total))
plot_counts2 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")
#Convert all Variants below 2 to other Variants
plot_counts <- plot_counts %>% mutate(Variant = case_when(Total <= 10 ~ "Other Variants", TRUE ~ as.character(Variant)))


#Plot the summarized table
plot_counts1 <- plot_counts %>% filter(Variant != "Other Variants")
plot_counts2 <- plot_counts %>% filter(Variant == "Other Variants")
plot_counts3 <- plot_counts2 %>% group_by(Variant) %>% summarise(across(Total:`<NA>`, sum))
plot_counts4 <- dplyr::bind_rows(plot_counts1,plot_counts3)
plot_counts5 <- plot_counts4 %>% summarise(across(Total:`<NA>`, sum)); rownames(plot_counts5)[rownames(plot_counts5) == "1"] <- "Grand Total"
plot_counts6 <- rownames_to_column(plot_counts5, "Variant")
plot_counts7 <- dplyr::bind_rows(plot_counts4,plot_counts6) #%>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021", "Total")# "Jul, 2021",
plot_counts7 %>%  knitr::kable(caption = "Overall gisaid Variants detected in Ghana")


plot_counts7 <- plot_counts4 #%>% select("Variant", "Mar, 2020", "May, 2020","Jun, 2020", "Jul, 2020", "Aug, 2020", "Sep, 2020", "Oct, 2020", "Nov, 2020",  "Dec, 2020",  "Jan, 2021", "Feb, 2021", "Mar, 2021", "Apr, 2021", "May, 2021", "Jun, 2021", "Jul, 2021", "Aug, 2021", "Sep, 2021")# "Jul, 2021",
row.names(plot_counts7) <- plot_counts7$Variant; #plot_counts7$Variant <- NULL;
plot_counts8 <- plot_counts7 %>% select(-c("Variant"))
plot_counts8 <- as.matrix(plot_counts8)
data3d <- prop.table(plot_counts8, 2)
data3d <- as.data.frame(data3d)

#Generate plots
plot_counts <- plot_counts %>% rename("Unknown" = `<NA>`)

p1 <- plot_counts %>%select(-c('Total')) %>% reshape2::melt(id.vars = 'Variant') %>%
  ggplot(aes(variable, value)) + geom_bar(aes(fill = Variant), position = 'fill', stat = 'identity', show.legend = TRUE) +
  #scale_fill_manual(name = 'Variant(Local)', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
    scale_fill_manual(name = 'Variant', values = c("B.1.617.2 (Delta)" = "#2484c1", "B.1.1.7 (Alpha)" = "chartreuse3",
                                                  "B.1.1.318" = "plum4","B.1.1" = "salmon4","B.1.525 (Eta)" = "red4",
                                                  "B.1.1.359" = "yellow3", "B.1"= "darkorange","B.1.623"="sienna3", 
                                                  "A" = "lightgoldenrod4","B.1.1.485" = "seagreen1",
                                                  "B.1.351 (Beta)" = "blue", "A.11"= "darkolivegreen4",
                                                  "Other Variants" = "darkgreen")) +
  scale_y_continuous(name = 'Percentage', labels = scales::percent, expand = c(0.01,0)) + theme_bw() + coord_flip() +
  #ggtitle("B) Dynamics of gisaid Variants from 2020 to 2021-Locals") +
  theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt')); p1
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Variant_distribution - Gender.svg', p1, width = 6, height = 3)
```


Plotting heatmap for mutations
```{r, echo=FALSE, warning=FALSE, comment=TRUE}

library(dplyr)
library(splitstackshape)
library(gtools)

#data5 <- data3 %>% separate(date1, c("Country","Location", "Year", "Date Collection", "Date Submitted"), sep = "([/|])")


#Select the relevant columns
data <- MetaPangoNext_Local %>% select(c("Virus.name", "Travel_info","Curated.Year.b", "Curated.month.year", "Curated.Region", "Variant", "aaSubstitutions", "substitutions"))

#Use the splitstackshape package to split the columns
data2 <- concat.split.multiple(
  data, split.cols = c("substitutions", "aaSubstitutions"),
  seps = ",", direction = "long")

#generate a column with gene names
data2$genes <- data2$aaSubstitutions
data2$genes <- gsub(":.*","",data2$genes, perl = TRUE)

#generate a column with just amino acid mutations
data2$aaMutation <- data2$aaSubstitutions
data2$aaMutation <- gsub(".*:","",data2$aaMutation, perl = TRUE)

###Plot for all samples.. most abundant mutations

##Now you can extract it and plot it
plot_counts <- data2 %>%  dplyr::group_by(aaSubstitutions, Variant) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Variant, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("aaSubstitutions", "Total", dplyr::everything())) %>% arrange(desc(Total))

#Convert all aaMutation below frequency of 10 to other aaMutation
plot_counts <- plot_counts %>% mutate(aaSubstitutions = case_when(Total <= 100 ~ "Others (n<100)",TRUE ~ as.character(aaSubstitutions)))

#Summarize the Variants
plot_counts1 <- plot_counts %>% filter(aaSubstitutions != "Others (n<100)") %>% select(c("aaSubstitutions", "Total")) %>% dplyr::filter(aaSubstitutions != "NA") %>% arrange(desc(Total))
plot_counts1 <- plot_counts1[order(as.numeric(gsub("([A-Z])", "", plot_counts1$aaSubstitutions, perl = T)), na.last=FALSE) , ]
library(forcats)
#plot_counts1$aaSubstitutions <- fct_inorder(plot_counts1$aaSubstitutions)

#generate a column with just amino acid mutations
plot_counts1$gene <- plot_counts1$aaSubstitutions
plot_counts1$gene <- gsub(":.*","",plot_counts1$gene, perl = TRUE)
plot_counts1 %>% knitr::kable(caption = "Highest Mutations frequency")

##Plot the most frequent mutations
p1 <- ggplot(plot_counts1, aes(x=aaSubstitutions, y=Total, fill=gene)) + 
  geom_col() + 
  #facet_wrap(~gene, scales = "free_y", ncol = 4) +
  theme_USGS_box() +
  scale_fill_manual(name = 'gene', values = custom_colors$discrete) + coord_cartesian(clip = 'off') +
  scale_y_continuous(limits = c(0, 1000), breaks = seq(0, 1000, by = 200)) +
  #geom_segment(aes(x=aaSubstitutions, xend=aaSubstitutions,  y=0, yend=Total)) + 
  #theme(axis.text.x = element_text(angle=65, vjust=0.6)) +
 theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 1, b = 1, l = 1, unit = 'pt')); p1
#p1 + coord_flip()
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Most abundant mutations  all genes.svg', width = 15, height = 6)

##Select spike protein and VOCs

#now you can easily filter specific group them
data3 <- data2 %>% dplyr::filter(genes == "S") %>% dplyr::filter(Variant == "B.1.1.7 (Alpha)" |Variant == "B.1.351 (Beta)" |Variant == "B.1.617.1 (Kappa)" |Variant == "B.1.1" |Variant == "B.1" |Variant == "A.27" |Variant == "A.23.1" |Variant == "B.1.617.2 (Delta)" |Variant == "B.1.525 (Eta)"  |Variant == "B.1.526 (Iota)" |Variant == "B.1.1.318" | Variant == "AY.* (Delta Plus)"); dim(data3)

##Now you can extract it and plot it
plot_counts <- data3 %>%  dplyr::group_by(aaMutation, Variant) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Variant, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("aaMutation", "Total", dplyr::everything())) %>% arrange(desc(Total))
#plot_counts %>% knitr::kable(caption = "Gisaid Variants by Regions - Local samples (2021)")
#Convert all aaMutation below frequency of 10 to other aaMutation
plot_counts <- plot_counts %>% mutate(aaMutation = case_when(Total <= 2 ~ "Others (n<2)",TRUE ~ as.character(aaMutation)))

#Summarize the Variants
plot_counts1 <- plot_counts %>% filter(aaMutation != "Others (n<2)")
plot_counts <- as.data.frame(plot_counts1)
row.names(plot_counts) <- plot_counts$aaMutation; plot_counts$aaMutation <- NULL; plot_counts$Total <- NULL;
plot_counts[] <- lapply(plot_counts, function(x) ifelse(x>1, 1, x))
plot_counts <- rownames_to_column(plot_counts, "mutation")
plot_counts <- plot_counts[order(as.numeric(gsub("([A-Z])", "", plot_counts$mutation, perl = T)), na.last=FALSE) , ]
rownames(plot_counts) <- plot_counts$mutation
plot_counts <- plot_counts %>% dplyr::select(-c("mutation"))
plot_counts %>% knitr::kable(caption = "Gisaid Variants by Regions - Local samples (2021)")

p1 <- pheatmap(plot_counts,  color = colorRampPalette((brewer.pal(n = 6, name = "Purples")))(100), cluster_rows = F,
         cluster_cols = F, clustering_distance_rows = "correlation",  treeheight_row = 0, treeheight_col = 0); p1

ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Spike_mutations.svg', p1, width = 4, height = 12)

##Now you can extract it and plot it
plot_counts <- data3 %>%  dplyr::group_by(aaMutation, Variant) %>% dplyr::summarize(count = n()) %>% 
  tidyr::spread(Variant, count, fill = 0) %>% dplyr::ungroup() %>% dplyr::mutate(Total = rowSums(.[c(2:ncol(.))])) %>% dplyr::select(c("aaMutation", "Total", dplyr::everything())) %>% arrange(desc(Total))
plot_counts %>% knitr::kable(caption = "Gisaid Variants by Regions - Local samples (2021)")
#Convert all aaMutation below frequency of 10 to other aaMutation
plot_counts <- plot_counts %>% mutate(aaMutation = case_when(Total <= 2 ~ "Others (n<2)",TRUE ~ as.character(aaMutation)))

#Summarize the Variants
plot_counts1 <- plot_counts %>% filter(aaMutation != "Others (n<2)") %>% select(c("aaMutation", "Total")) %>% 
  dplyr::filter(aaMutation != "NA") %>% arrange(desc(Total))
plot_counts1 <- plot_counts1[order(as.numeric(gsub("([A-Z])", "", plot_counts1$aaMutation, perl = T)), na.last=FALSE) , ]
library(forcats)
plot_counts1$aaMutation <- fct_inorder(plot_counts1$aaMutation)

##Plot the most frequent mutations
p1 <- ggplot(plot_counts1, aes(x=aaMutation, y=Total)) + 
  geom_point(size=3) + 
  theme_USGS_box() +
  geom_segment(aes(x=aaMutation, xend=aaMutation,  y=0, yend=Total)) + 
  #theme(axis.text.x = element_text(angle=65, vjust=0.6)) +
 theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 1, b = 1, l = 1, unit = 'pt')); p1
#p1 + coord_flip()
ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Most abundant mutations_ spike.svg', width = 12, height = 3)



```

Global & Ghanaian COVID-19 Trends 

```{r, echo=FALSE, warning=FALSE, comment=FALSE}
who <- read_csv("/Users/collinsmisita/manuscript/017_Global_Data/WHO-COVID-19-global-data.csv")


Africa <- who %>%
  filter(WHO_region == "AFRO")

west_africa_list <- c("Nigeria","Ghana","Cameroon","Niger","Burkina Faso", 
                      "Mali","Senegal","Guinea","Benin","Togo","Sierra Leone", 
                      "Liberia","Mauritania","Gambia","Guinea Bissau", 
                      "Cape Verde")

west_africa <- who %>%
  filter(Country == west_africa_list) %>%
  select(Date_reported,Country,New_cases)

West_Africa <- west_africa %>% rename(West_Africa = New_cases)
  
Ghana <- who %>%
  filter(Country == "Ghana")

newCases <- 
  ggplot() +
  #geom_line(aes(x = who$Date_reported, y = log(who$New_cases),color = "New_cases_worldwide")) +
  geom_line(aes(x = Africa$Date_reported, y = log(Africa$New_cases),color = "New_cases_Africa")) +
  #geom_line(aes(x = West_Africa$Date_reported, y = log(West_Africa$West_Africa),color = "New_cases_West_Africa")) +
  geom_line(aes(x = Ghana$Date_reported, y = log(Ghana$New_cases), color = "New_cases_Ghana")) +
  scale_x_date(date_breaks = "1 month", date_labels = "%d %b") +
  scale_y_continuous(labels = scales::comma) +
  labs( x = "Dates", y = "Number of new cases",
        title = "") +
    theme_bw() +
  theme(legend.position = 'top', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 20, r = 0, b = 0, l = 0, unit = 'pt')) +
    scale_color_manual(name="", values = c("New_cases_worldwide" = "tan", "New_cases_Africa"="steelblue", "New_cases_West_Africa"="purple", "New_cases_Ghana"="orange")) ; newCases

ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/newCase_all.png', newCases, width = 5, height = 4)

```

```{r, echo=FALSE, warning=FALSE, comment=FALSE}

#Select the relevant columns
data <- MetaPangoNext_Local %>% select(c("Virus.name", "Travel_info","Curated.Year.b", "Collection_date", "Curated.month.year", "totalSubstitutions", "totalAminoacidSubstitutions", "Curated.Region", "Variant", "aaSubstitutions", "substitutions")) %>% filter(!is.na(Collection_date)) %>% filter(Collection_date != "Unknown")

#now you can easily filter specific group them
data3 <- data %>% dplyr::filter(Variant == "B.1.1.7 (Alpha)" |Variant == "B.1.351 (Beta)" |Variant == "B.1.617.1 (Kappa)" |Variant == "B.1.1" |Variant == "B.1" |Variant == "A.27" |Variant == "A.23.1" |Variant == "B.1.617.2 (Delta)" |Variant == "B.1.525 (Eta)"  |Variant == "B.1.526 (Iota)" |Variant == "B.1.1.318"); dim(data3)

#Fever
metadata2 <- data %>% select(c(totalSubstitutions, Curated.month.year, Variant)) %>% filter(!is.na(Curated.month.year))
p2 <- ggplot(metadata2, aes(x=Variant, y=totalSubstitutions)) +
  geom_count(width=0.08, color="black", position=position_dodge(width = 0.9)) + theme_bw() + 
  #scale_fill_manual(values=c("red", "blue")) + theme(text=element_text(size=12)) +
  theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), 
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1)) +
  labs(x="Variant ", size = 10) + labs(y="Number of mutations", size = 10) + coord_flip(); p2

ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/mutation_Frequency_Perlineage.svg', p2, width = 6, height = 8)


```


PHYLOGENETICS


A) Phylogeny of top 5 variants

```{r, echo=FALSE,warning=FALSE}
#############################################################

#phylogenetics <- MetaPangoNext_QC %>% dplyr::filter(waccbipQc == "  Good" | waccbipQc == " AcceptableForSubmission") %>% filter(Travel_info !=   "Nigeria") %>% filter(Host_GIS == "Human") %>% filter(is.na(waccbipQc1)); dim(phylogenetics) # filter(!is.na(Collection.date)) %>%

 phylogenetics <- MetaPangoNext_QC %>% filter(Travel_info ==   "Local")  
# 
# Nextstrain <- phylogenetics %>% select(c("Collection.date", "Collection_date", "strain", "virus", "Accession.ID", "Accession", "region", "country", "division", "District", "region_exposure",
# "country_exposure", "division_exposure", "segment", "alignmentEnd", "age", "Gender"))
# 
# Nextstrain2 <- Nextstrain %>% rename(date_submitted = Collection.date) %>% 
#   rename(date = Collection_date) %>% rename(location = District) %>% rename(length = alignmentEnd) %>% 
#   rename(sex = Gender)  %>% rename(gisaid_epi_isl = Accession.ID)  %>% rename(genbank_accession = Accession)
# 
# #library(lubridate, warn.conflicts = FALSE)
# 
# Nextstrain2$age <- as.character(Nextstrain2$age)
# 
# #lubridate::guess_formats(Nextstrain$date_submitted, orders = "dmy", local = "C")
# 
# Nextstrain2$date <- dmy(Nextstrain2$date)
# Nextstrain2$date_submitted <- dmy(Nextstrain2$date_submitted)
# 
# ref <- readr::read_tsv(file = "/Users/collinsmisita/manuscript/015_nextstrain/ncov/data/references_metadata.tsv", col_types = cols())
# compare_df_cols(Nextstrain2, ref, bind_method = "rbind")
# Nextstrain_ref <- dplyr::bind_rows(Nextstrain2, ref)
# 
# readr::write_tsv(Nextstrain_ref, file='/Users/collinsmisita/manuscript/015_nextstrain/ncov/data/Nextstrain_waccbip_ref.tsv')
# 
# inclusion_samples <- phylogenetics %>%dplyr::select(c('strain'))
# readr::write_tsv(inclusion_samples, file='/Users/collinsmisita/manuscript/015_nextstrain/Waccbip_Data/inclusion_Data.tsv', col_names = FALSE)
# 
# write.table(inclusion_samples, "/Users/collinsmisita/manuscript/007_merged_fasta_file/subset_fasta/ids2.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)

#write.table(Nextstrain_ref, file='/Users/collinsmisita/manuscript/015_nextstrain/ncov/data/example_metadata.tsv', quote=FALSE, sep='\t', col.names = FALSE, row.names = FALSE)



tree<-read.newick('/Users/collinsmisita/ncov5/results/global/tree.nwk')

#tree_raw<-read.newick('/Users/collinsmisita/manuscript/015_nextstrain/ncov/results/global/tree_raw.nwk')


meta <- as.data.frame(tree$tip.label)

#Merge the datasets
metadata_df <-merge(meta,phylogenetics, by.x ="tree$tip.label", by.y = "strain", all.x = TRUE)
row.names(metadata_df) <- metadata_df$`tree$tip.label`

metadata_df <- metadata_df %>% rename(date_submitted = Collection.date) %>% rename(variant = Variant) %>%
  rename(date = Collection_date) %>% rename(location = District) %>% rename(length = alignmentEnd) %>%
  rename(sex = Gender)  %>% rename(gisaid_epi_isl = Accession.ID)  %>% rename(genbank_accession = Accession)


metadata_df$date <- dmy(metadata_df$date)
metadata_df$days<-as.Date(cut(metadata_df$date,breaks = "day",start.on.monday = FALSE))
metadata_df$date2<-as.Date(cut(metadata_df$date,breaks = "2 weeks",start.on.monday = FALSE))
metadata_df$date3<-as.Date(cut(metadata_df$date,breaks = "1 month",start.on.monday = FALSE))


custom3<-c('antiquewhite2',"tan4",'peachpuff3','palegreen3','dodgerblue1','hotpink2','mediumorchid3','blue3','skyblue1','purple4',
           'mediumpurple','darkseagreen4','cadetblue3','thistle3','deeppink3',
           #custom2<-c("darkolivegreen",'darkseagreen3','darkseagreen2','antiquewhite2','peachpuff3','tan4',
           'red3','grey30','darkolivegreen','slategray1','plum2',
           'white','white')

p<-ggtree(tree, as.Date=TRUE, mrsd="2021-09-08", color='black',size=0.2) + theme_tree2()+
  expand_limits(y = 2)+ theme_tree2() +
  theme(axis.text.x = element_text(size=10,angle=90)) +
  geom_nodepoint(color="grey82", alpha=1/4, size=0.5) #+
  #geom_tiplab(align=FALSE, linesize=.1)


colors <- c("variant" = "blue")
p <- p %<+% metadata_df +
  geom_tippoint(aes(
    subset=(region=='Africa')),fill='white',size=3,align=F,stroke=0.2,color='grey36',shape=21)+
  geom_tippoint(aes(
    subset=(variant=='B.1.1.7 (Alpha)')),fill='chartreuse3',size=4, stroke=0.2,align=F, color='grey36',shape=21)+
  geom_tippoint(aes(
    subset=(variant=='Q.1')),fill='chartreuse3',size=4, stroke=0.2,align=F, color='grey36',shape=21)+
  geom_tippoint(aes(
    subset=(variant=='B.1.351 (Beta)')),fill='blue',size=4,align=F, stroke=0.2,color='grey36',shape=21)+
  geom_tippoint(aes(
    subset=(variant=='B.1.525 (Eta)')),fill='red4',size=4,align=F, stroke=0.2,color='grey36',shape=21)+ 
  geom_tippoint(aes(
    subset=(variant=='B.1.617.2 (Delta)')),fill='#2484c1',size=4, stroke=0.2,align=F, color='grey36',shape=21)+
    geom_tippoint(aes(
    subset=(variant=='B.1.1.318')),fill='plum4',size=4, stroke=0.2,align=F, color='grey36',shape=21)+
   scale_color_manual(values = colors) +
  scale_x_date(breaks = date_breaks("1 month")) +
   theme(legend.position = 'right', plot.title = element_text(hjust = 0.5), text = element_text(size = 12),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), plot.margin = margin(t = 10, r = 10, b = 10, l = 10, unit = 'pt')) #+ geom_tiplab()

p 

ggsave('/Users/collinsmisita/manuscript/009_merged_lineage_reports/Figures/Phylogeny.svg', p, width = 10, height = 18, units = "cm",limitsize = FALSE)

```

```{r}
# layout(1:2)
# plot(0,0,type="n",xlim=c(0,50),ylim=c(0,350),xlab="CT Values",ylab="Frequency",main="Distribution of CT Values")
# 
# target1 <- MetaPangoNext_Local %>%select(c('Target1', "Variant"))
# target1[target1 == ""] <- NA
# target1[target1 == "0"] <- NA
# target1b <- target1[complete.cases(target1),]
# target1b$Target1b <- as.numeric(target1b$Target1)
# p1 <- hist(target1b$Target1b,plot=FALSE)
# 
# target2 <- MetaPangoNext_Local %>%select(c('Target2', "Variant"))
# target2[target2 == ""] <- NA
# target2[target2 == "0"] <- NA
# target2b <- target2[complete.cases(target2),]
# target2b$Target2b <- as.numeric(target2b$Target2)
# p2 <- hist(target2b$Target2b,plot=FALSE)
# 
# plot(p1,col="red",density=10,angle=135,add=TRUE)
# plot(p2,col="blue",density=10,angle=45,add=TRUE)
# 
# legend('topleft',c('p1','p2'),fill = c("red", "blue"), bty = 'n',border = NA)
# ##save the histogram
# ##########################################################33
Regression <- MetaPangoNext_Local %>%select(c("VariantType", "age", "Gender","agegroup","agegroup2","Curated.month.year", "Curated.Region"))

Regression[Regression == "Unknown"] <- NA
Regression$age <- as.numeric(Regression$age)

library(gtsummary)
require(foreign)
require(nnet)
require(ggplot2)
require(reshape2)

# tab1 <- Regression %>%
#   tbl_summary(by = "VariantType") %>%
#   add_p() %>%
#    #add_p(pvalue_fun = ~style_pvalue(.x, digits = 2)) %>%
#    add_overall() %>%
#    add_n() %>%
#    modify_header(label ~ "**Variable**") %>%
#    modify_spanning_header(c("stat_1", "stat_2") ~ "**Study Group**") %>%
#    modify_footnote(
#      all_stat_cols() ~ "Median (IQR) or Frequency (%)"
#    ) %>%
#    modify_caption("**Table 1. Patient Characteristics**") %>%
#    bold_labels()
# tab1
# 
# library(rstatix)
# 
# 
# fisher.test(table(Regression$VariantType, Regression$agegroup), simulate.p.value = TRUE, B = 1e5)
# fisher.test(table(Regression$VariantType, Regression$Gender), simulate.p.value = TRUE, B = 1e5)
# fisher.test(table(Regression$VariantType, Regression$Curated.month.year), simulate.p.value = TRUE, B = 1e5)
# fisher.test(table(Regression$VariantType, Regression$Curated.Region), simulate.p.value = TRUE, B = 1e5)

# xtab <- table(Regression$Gender, Regression$VariantType)
# library(fifer)
# chisq_test(xtab)
# chisq.post.hoc(xtab, test='chisq.test')
# 
# 
# xtab <- table(Regression$VariantType, Regression$agegroup2)
# chisq.test(xtab)
# chisq.post.hoc(xtab, test='chisq.test')

# write.csv(Regression, "../../../007_Waccbip_Manuscripts/Covid-2021_manuscript/NatureComm/ENA_DATA_Submission/Regression.csv")

```


